libraries/src/Mail/MailTemplate.php

```diff
@@ -485,29 +486,20 @@ public static function deleteTemplate($key)
 	 *
 	 * @since   4.0.0
 	 */
-	protected function setAttachmentName($file, $name)
+	protected function getAttachmentName($file, $name)
 	{
-		// If no name is given use the basename of the file to be attached
+		// If no name is given, do not process it further
 		if (!trim($name))
 		{
-			return basename($file);
+			return '';
 		}
 
-		// Get file extension of the file to be attached
-		$extensionFile = pathinfo($file, PATHINFO_EXTENSION);
-
-		// If the file has no extension there is nothing to do with the name
-		if (!$extensionFile)
-		{
-			return $name;
-		}
-
-		// Make sure to attach the file with the same extension as the attached file has (case-insensitive)
-		if (strtolower(pathinfo($name, PATHINFO_EXTENSION)) !== strtolower($extensionFile))
+		// The given name use a different file extension, something could not be right, ignore it
+		if (File::getExt($file) !== File::getExt($name))
 		{
-			return $name . '.' . $extensionFile;
+			return '';
 		}
 
-		return $name;
+		return File::makeSafe($name);
```

Comment:
Not sure about the makeSafe .. it could do more than desired, e.g. forbid Greece or Cyrillic characters. I have to check.